psq/worker.py

```diff
@@ -42,7 +45,12 @@ def _safe_dequeue(self):
             wait_exponential_multiplier=1000, wait_exponential_max=10000,
             retry_on_exception=lambda e: not isinstance(e, KeyboardInterrupt))
         def inner():
-            return self.queue.dequeue(max=self.tasks_per_poll, block=True)
+            try:
+                return self.queue.dequeue(max=self.tasks_per_poll, block=True)
+            except errors.GaxError as e:
```

Comment:
Yeah we are going to just swallow the `DEADLINE_EXCEEDED`. In some sense, that is the appropiate response and the JSON-over-HTTP API is the one getting it wrong on a timeout.